torchvision/csrc/cpu/image/readjpeg_cpu.cpp

```diff
@@ -100,15 +101,35 @@ torch::Tensor decodeJPEG(const torch::Tensor& data) {
 
   // read info from header.
   jpeg_read_header(&cinfo, TRUE);
+
+  int current_channels = cinfo.num_components;
+
+  if (components > 0 && components != current_channels) {
+    switch (components) {
+      case 1: // Gray
+        cinfo.out_color_space = JCS_GRAYSCALE;
+        break;
+      case 3: // RGB
+        cinfo.out_color_space = JCS_RGB;
+        break;
+      default:
+        jpeg_destroy_decompress(&cinfo);
+        TORCH_CHECK(false, "Invalid number of output channels.");
+    }
+
+    jpeg_calc_output_dimensions(&cinfo);
+  } else {
+    components = current_channels;
+  }
+
   jpeg_start_decompress(&cinfo);
 
   int height = cinfo.output_height;
   int width = cinfo.output_width;
-  int components = cinfo.output_components;
 
-  auto stride = width * components;
-  auto tensor = torch::empty(
-      {int64_t(height), int64_t(width), int64_t(components)}, torch::kU8);
+  int stride = width * components;
```

Comment:
nit: can we make this a `int64_t`, so that we don't have more warnings during compilation?

Reply:
This is intentional to maintain the same behaviour as before. See the original code: 
- The components and width were integers and so stride was int.
- Since channels/components now comes from Python it's int64_t, so here I define stride as integer explicitly.

Thoughts?